### PR TITLE
test(text-field): Add screenshot tests

### PR DESCRIPTION
**Pull request by williamernest**
Adds screenshot tests for text field

### CODE REVIEW

**Comment by kfranqueiro** [comment]
FYI I think helper text moved since your recent merge w/ master due to https://github.com/material-components/material-components-web/pull/3297, which requires removing leading/trailing space from the helper text element (it's unfortunate but we couldn't find an alternative that worked in all browsers).

**Comment by acdvorak** [comment]
[`textarea-invalid-focused.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/textarea-invalid-focused.html) doesn't show focused state because the `textarea` CSS selector relies on `:focus`.

Are you keeping this as a placeholder page for when we bring `<textarea>` up to spec?

---

Some sets of pages look identical, despite having slightly different markup.

Am I correct in assuming that the pages below are testing that `.mdc-text-field-helper-text--validation-msg` does _not_ turn red if the textfield is _not_ "invalid"?

*   [`baseline-helper-text-persistent-validation-msg.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/baseline-helper-text-persistent-validation-msg.html) \
    vs. \
    [`baseline-helper-text-persistent.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/baseline-helper-text-persistent.html)

*   [`baseline-helper-text-validation-msg.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/baseline-helper-text-validation-msg.html) \
    vs. \
    [`baseline-helper-text.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/baseline-helper-text.html)

*   [`disabled-helper-text-persistent-validation-msg.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/disabled-helper-text-persistent-validation-msg.html) \
    vs. \
    [`disabled-helper-text-persistent.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/disabled-helper-text-persistent.html)

*   [`disabled-helper-text-validation-msg.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/disabled-helper-text-validation-msg.html) \
    vs. \
    [`disabled-helper-text.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/disabled-helper-text.html)

*   [`focused-helper-text-persistent-validation-msg.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/focused-helper-text-persistent-validation-msg.html) \
    vs. \
    [`focused-helper-text-persistent.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/focused-helper-text-persistent.html)

*   [`invalid-focused-helper-text-persistent-validation-msg.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/invalid-focused-helper-text-persistent-validation-msg.html) \
    vs. \
    [`invalid-focused-helper-text-validation-msg.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/invalid-focused-helper-text-validation-msg.html)

*   [`textarea-invalid-focused.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/textarea-invalid-focused.html) \
    vs. \
    [`textarea-invalid.html`](https://storage.googleapis.com/mdc-web-screenshot-tests/travis/2018/08/06/20_51_26_951/spec/mdc-textfield/classes/textarea-invalid.html)

File: test/screenshot/spec/mdc-textfield/fixture.scss [hunk 1]
```diff
@@ -19,5 +19,10 @@
 
 .test-cell--textfield {
   width: 301px;
-  height: 101px;
+  height: 120px;
```
Comment (acdvorak) [inline test/screenshot/spec/mdc-textfield/fixture.scss]: `height: 121px` (so the cell ends on a gridline)

File: test/screenshot/spec/mdc-textfield/fixture.scss [hunk 2]
```diff
@@ -19,5 +19,10 @@
...
+}
+
+.test-cell--textarea {
+  width: 301px;
+  height: 175px;
```
Comment (acdvorak) [inline test/screenshot/spec/mdc-textfield/fixture.scss]: `height: 181px` (so the cell ends on a gridline)

**Comment by williamernest** [comment]
@acdvorak I think I got a little crazy with the `Ctrl+C` & `Ctrl+V`. I'll remove the text field invalid focused page. 

The `validation-msg` ones are to test the validation logic only turning the message red if the field is invalid.

**Comment by kfranqueiro** [comment]
SGTM, thanks Andy and Will!

**Pull request merged**